docs/developer-resources/contractkit/migrating-to-contractkit-v2.md

```diff
@@ -0,0 +1,260 @@
+---
+title: Migrating to ContractKit v2.0
+description: How to migrate from v1 to v2 ContractKit suite of packages and make use of their latest features.
+slug: /developer-guide/contractkit/migrating-to-contractkit-v2
+---
+
+
+How to migrate from v1 to v2 of the Celo SDK suite of packages and make use of their latest features.
+
+___
+
+## Why v2?
+
+### Bundlesize
+
+The primary motivation in creating v2 was reduced bundlesize and increased real modularity. The massive package size for `@celo/contractkit` has been an elephant in the room and source of dissonance for looking to build mobile first dApps. As of 1.5.2 bundlephobia list the minified size at 3.7MB. 2.0.0 comes in at 1.7MB. still big yet we have a few more tricks. First the packages have been all marked as `sideEffects:false`, `kit` instance is no longer required to any classes in the contractkit package, and the introduction `MiniContractKit`.
+
+### Modularity
+
+#### `sideEffects:false`
+
+Tells your bundler it can safely only include the code that is explicitly used, reducing real world bundlesize
+
+#### `kit` no longer needed by everything
+
+In v1 Almost everything required a Kit to be passed to its constructor, effectively this meant it was impossible to use any of the classes in @celo/contractkit alone.
+
+In v2 AddressRegistry, Wrappers, WrapperCache, and more can all be constructed using mostly just a `Connection` (sometimes other arguments too).
+
+#### `MiniContractKit`
+
+The prize of no longer needing a full kit is that it became possible to create a slimed down minimal viable ContractKit.
+
+`MiniContractKit` provides a subset of ContractKit features with the same interface. For many dapps it will be a drop in opt-in change eg `import {newKit, ContractKit} from "@celo/contractkit/lib/mini/kit` It drops weight by only including access to `Accounts, GasPriceMinimum, StableToken*, Exchange* and GoldToken` wrappers and contracts. It can `setFeeCurrency`, look up info about the current account, and like full Contractkit it delegates most functionality to `connection`.
+
+## Get Started
+
+Upgrade The packages your project uses to the latest (in this case release beta).  For example, with ContractKit and Celo utils:
+
+`yarn add @celo/contractkit@beta @celo/utils@beta`
+
+If you are directly importing any other @celo/* packages, upgrade them as well for full benefit.
+
+If you need them, append `@celo/phone-utils@beta` `@celo/cryptographic-utils@beta`.
+
+*(see section on breaks in [@celo/utils](#celoutils) to know if you need them)*
```

Comment:
is this referring to breaking changes? I don't see any info about breaking changes in the utils section

Reply:
thats the largest part.

Maybe its not clear since i used to new package names to group the breaking changes?

Reply:
Ok I get it now, thanks